include/boost/graph/minimum_degree_ordering.hpp

```diff
@@ -317,7 +319,8 @@ namespace boost {
         // Initialize degreelists.  Degreelists organizes the nodes
         // according to their degree.
         for (boost::tie(v, vend) = vertices(G); v != vend; ++v) {
-          put(degree, *v, out_degree(*v, G));
+          put(degree, *v, d = out_degree(*v, G));
+          if (0 < d) has_no_edges = false;
           degreelists.push(*v);
         }
```

Comment:
For simplicity, I would prefer the assignment `d = ...` to happen before the `put` rather than inline, and I would move the declaration of `d` inside that loop too. It's just a preference for scope and simplicity, nothing serious.